docs/source/using-federation/apollo-subgraph-setup.mdx

```diff
@@ -0,0 +1,351 @@
+    me: User
+  }
+
+  type User @key(fields: "id") {
+    id: ID!
+    username: String
+  }
+`;
+
+const resolvers = {
+  Query: {
+    me() {
+      return { id: '1', username: '@ava' };
+    },
+  },
+  User: {
+    __resolveReference(user, { fetchUserById }) {
+      return fetchUserById(user.id);
+    },
+  },
+};
+
+const server = new ApolloServer({
+  schema: buildSubgraphSchema({ typeDefs, resolvers }),
+});
+
+const { url } = await startStandaloneServer(server);
+console.log(`🚀  Server ready at ${url}`);
+```
+
+</MultiCodeBlock>
+
+## Custom directives in subgraphs
+
+The method for defining custom directives differs slightly for a federated graph, and it also depends on the [version of Apollo Server](../previous-versions) you're using.
+
+> ⚠️ **Important considerations**
+>
+> Before you use directives in a federated graph, make sure to consider the following:
+>
+> - Custom directives are _not_ included in your graph's composed supergraph schema. The composition process strips all subgraph directives. Only a given subgraph is aware of its own directives.
+> - Because directives are specific to individual subgraphs, it's valid for different subgraphs to define the _same_ directive with _different_ logic. Composition does not detect or warn about such inconsistencies.
+> - If multiple subgraphs can resolve a particular field, each subgraph should almost always apply the exact same set of custom directives (with the exact same accompanying logic) to that field. Otherwise, the behavior of that field might vary depending on _which_ subgraph resolves it.
+
+### Directives in Apollo Server 4.x and 3.x
+
+Apollo Server versions 3 and 4 do not provide _built-in_ support for custom directives, but you can install certain `@graphql-tools` libraries to enable support. To get started with these libraries in Apollo Server, first read [Creating schema directives](/apollo-server/v3/schema/creating-directives/).
+
+As the linked article describes, in Apollo Server 3 and 4 you define a **transformer function** for each of your subgraph schema's custom directives.
+
+To apply transformer functions to your executable subgraph schema, you first _generate_ the subgraph schema with `buildSubgraphSchema` as usual:
+
+<MultiCodeBlock>
+
+```ts
+let subgraphSchema = buildSubgraphSchema({ typeDefs, resolvers });
+```
+
+</MultiCodeBlock>
+
+But instead of passing the result directly to the `ApolloServer` constructor, you first apply all of your transformer functions to it:
+
+<MultiCodeBlock>
+
+```ts
+// Transformer function for an @upper directive
+subgraphSchema = upperDirectiveTransformer(subgraphSchema, 'upper');
+```
+
+</MultiCodeBlock>
+
+After applying all transformer functions, you provide your final subgraph schema to the `ApolloServer` constructor as usual:
+
+<MultiCodeBlock>
+
+```ts
+const server = new ApolloServer({
+  schema: subgraphSchema,
+  // ...other options...
+});
+```
+
+</MultiCodeBlock>
+
+### Directives in Apollo Server 2.x
```

Comment:
I'd prefer not to have any docs about AS2 features inside the AS4 docs! However, I'm pretty sure that the techniques that work with AS3/AS4 work with AS2 as well (since they are independent of Apollo Server).

At the very least, let's delete the AS2 stuff and just talk about using graphql-tools (though the sample code here doesn't really tell you how to write `upperDirectiveTransformer`...)

But actually, I feel like we could delete the entire "Custom directives in subgraphs" section from this file, and add a short section about directives in subgraphs (mostly just the "important considerations") to `schema/directives` instead? We don't actually document transformer stuff in the AS docs anymore anyway though...

Reply:
1. I'll remove those Apollo Server 2 directions (which feels reasonable with `graphql-tools` working for all versions). 
2. I removed the content from the "Custom directives in subgraphs" section and added a "Custom Directives" section (with an accompanying "In subgraphs" section) to the "Directives" article! I like this arrangement much better—thanks for the great idea! ✨